spec/components/button_spec.rb

```diff
@@ -22,6 +22,18 @@ def component_name
     assert_select ".govuk-button", text: "Submit"
   end
 
+  it "renders with the correct types" do
+    render_component(text: "Submit")
+    assert_select ".govuk-button[type=submit]", text: "Submit"
+
+    render_component(text: "Link", href: "#")
+    assert_select "a.govuk-button", text: "Link"
+    assert_select "a.govuk-button[type=submit]", false
```

Comment:
We may also want a test that checks if you pass a `href` and a `type`, that the type isn't applied (to confirm that type isn't applied if the button is actually a link).

Reply:
Good point, will add.

Reply:
I'd be tempted to split this into multiple tests so if one part fails it's clearer exactly where it's gone wrong. But not a blocker